.github/workflows/dependency_modification_check.yml

```diff
@@ -0,0 +1,18 @@
+name: Check no dependencies were modified
+
+on:
+  pull_request
+
+jobs:
+  dependency_changed_check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: "Check Cargo.lock not in changeset" 
+        run: |
+          git fetch origin
+          git diff origin/$GITHUB_BASE_REF.. --name-only| ( ! grep "Cargo.lock")
```

Comment:
nit: may want to add the `-q` so grep doesn't print, but irrelevant if we never see the output

Reply:
Mh, you can see the output by clicking on "Details" below. It tells you what Cargo.lock file caused the failure, although I guess that's also clearly visible from the "Files Changed" view, lol.